mysql.py

```diff
@@ -36,3 +36,36 @@ def fix_replication_from_slow_query_log_after_upgrade():
     run_mysql_command("START SLAVE;")
     run_mysql_command("SET GLOBAL slow_query_log = 'ON';")
     run_mysql_command("show slave status\G;")
+
+
+@task
+def setup_slave_from_master(master):
+    """
+    Sets up a slave from a master by taking a dump from the master,
+    copying it to the slave and then restoring the dump.
+
+    Usage: fab environment -H mysql-slave-1.backend mysql.setup_slave_from_master:'mysql-master-1.backend'
+    """
+    if len(env.hosts) > 1:
+        print 'This job is currently only setup to run against one slave at a time'
+
+    with settings(host_string=master):
+        # The use of `--master-data` here implies `--lock-all-tables` per the
+        # MySQL reference manual: http://dev.mysql.com/doc/refman/5.1/en/mysqldump.html#option_mysqldump_master-data
+        run('sudo -i mysqldump -u root --all-databases --master-data --add-drop-database > dump.sql')
+
+    with settings(host_string=master, forward_agent=True):
+        run('scp dump.sql {0}:~'.format(env.hosts[0]))
+
+    with settings(host_string=master):
+        run('rm dump.sql')
```

Comment:
On L55, you created the dump as `root`, but I think the upshot here will be that you're trying to remove the file as yourself, because the user used will be whatever is set in your `.fabricrc`. Maybe it would be better to run this as either `sudo('rm dump.sql')`, or if you'd rather not use sudo, then pass `-f` to `rm` as it stands.